code/game/objects/structures/holosign.dm

```diff
@@ -180,6 +180,39 @@
 	else
 		return ..()
 
+/obj/structure/holosign/barrier/detective
+	name = "investigation barrier"
+	desc = "An authoritive holographic barrier proclaiming a crime scene. Energy arcs off of it."
+	max_integrity = 20
+	icon_state = "holosign_det"
+	var/shockcd = 0
+
+/obj/structure/holosign/barrier/detective/bullet_act(obj/projectile/P)
+	take_damage(P.damage, BRUTE, MELEE, 1)
+	return BULLET_ACT_HIT
+
+/obj/structure/holosign/barrier/detective/proc/cooldown()
```

Comment:
Did you want to throw this after all the inherited procs, line 215, doesn't make much sense to have it live here.